orchagent/nhgorch.cpp

```diff
@@ -532,11 +637,8 @@ bool NextHopGroup::sync()
         return true;
     }
 
-    /*
-     * If the group is temporary, the group ID will be the only member's NH
-     * ID.
-     */
-    if (m_is_temp)
+    /* If the group is non-recursive with single member, the group ID will be the only member's NH ID */
+    if (!isRecursive() && (m_members.size() == 1))
```

Comment:
Do we need m_is_temp check?

Reply:
That'll be redundant as the current check will be true for "temp" nhgs as well.